src/amici.cpp

```diff
@@ -1424,7 +1794,11 @@ int workForwardProblem(UserData *udata, TempData *tdata, ReturnData *rdata, cons
                         }
                     } else {
                         if (udata->nx>0) {
-                            status = AMISolve(ami_mem, RCONST(udata->ts[it]), tdata->x, tdata->dx, &(tdata->t), AMICI_NORMAL);
+                            if (isinf(udata->ts[it])) {
+                                status = workSteadyStateProblem(udata, tdata, rdata, ami_mem, it);
```

Comment:
Thanks!
It would, yes. Preequlibration however, I wanted to do via a flag which can be
 * 0 - no preequilibration
 * 1 - preequilibration of states only
 * 2 - preequilibration of states and sensitivities
However, doing this directly in the simulation would keep the code simpler...
We should discuss that offline.